collectors/conn.go

```diff
@@ -64,6 +81,66 @@ func (n *NoopConn) Shutdown() {}
 
 // MonCommand returns the provided output string to NoopConn as is, making
 // it seem like it actually ran something and produced that string as a result.
-func (n *NoopConn) MonCommand(_ []byte) ([]byte, string, error) {
+func (n *NoopConn) MonCommand(args []byte) ([]byte, string, error) {
+	// Unmarshal the input command and see if we need to intercept
+	cmd := map[string]interface{}{}
+	err := json.Unmarshal(args, &cmd)
+	if err != nil {
+		return []byte(n.output), "", err
+	}
+
+	// Intercept and mock the output
+	switch prefix := cmd["prefix"]; prefix {
+	case "pg dump":
+		dc, ok := cmd["dumpcontents"].([]interface{})
+		if !ok || len(dc) == 0 {
+			break
```

Comment:
This will return "", "", nil -- might help to return an error here if `dumpcontents` is not in the right format?

Reply:
Yup. Need to check the presence of `dumpcontents` for robustness.